[audipras] iP

audipras opened the pull request:
### MAGA Chatbot

> "Despite the negative press covfefe." - Donald Trump ([source](https://www.reuters.com/article/world/trumps-covfefe-tweet-leaves-internet-guessing-idUSKBN18R27J/))

The release of this chatbot is yuuuuuge - like, YUUUGE. It's gonna solve **ALL** your task-related problems. Just trust me - its the best 👌. It has:

- Easy text interface
- Simple, easy commands
- ~~TRUMP~~ **_DRUMPF_** SPEECH

Features:

- [ ] Create tasks
- [ ] Mark tasks complete
- [ ] Create events and deadlines

People who endorse this app:

1. Me (Donald Trump)
2. The people of America
3. Crooked Kamala (Yes, even her!)

I support all jobs and occupations in America - yes, even our dear musicians and artists. Look at the beautiful art this bot has:

```
    private void printGreeting() {
        String logo = "  __  __                    \n"
                + " |  \\/  |  __ _   __ _   __ _  \n"
                + " | |\\/| | / _` | / _` | / _` | \n"
                + " | |  | || (_| || (_| || (_| || \n"
                + " |_|  |_| \\__,_| \\__, | \\__,_|  \n"
                + "                  |___/                           \n";
        System.out.println("Hello from\n" + logo +"\nI am THE best chatbot from the one and only" +
                " US of A trust me everyone says I'm the best. How can I help you serve the American people?" );
    }
```

Download this bot _nowww_ 👌 - Make America Great Again!!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
level 0

Commit:
level 1

Commit:
level 2

Commit:
level 3

Commit:
level 4

Commit:
level 5

Commit:
level 6

Commit:
bug fix

Commit:
Revert "bug fix"

This reverts commit 66ca9d21d31a2f634327259b45cf413fd730e046.

Commit:
git bug fix

audipras committed:
Merge branch 'nus-cs2103-AY2425S1:master' into master

audipras committed:
Create taskList class

audipras committed:
Tasklist class: fix delete and list errors

audipras committed:
Create TaskManager class handling save and load

audipras committed:
Add date time formatting

audipras committed:
Package task-related classes into task

dominic2412 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,80 @@
+import task.TaskList;
```
Parser is a good name for a class responsible for parsing input. It clearly describes its purpose. Well done here!

dominic2412 commented on src/main/java/Maga.java:
```diff
@@ -0,0 +1,40 @@
+    private void run() {
+        printGreeting();
+        TaskManager taskManager = new TaskManager();
+        TaskList taskList = initialiseBot(taskManager);
```
accurately describes the action it performs, which is good practice.

dominic2412 commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,110 @@
+    }
+
+    public void deleteTask(int taskNumber) {
+        Task tempTask = taskList[taskNumber];
```
tempTask is used to refer to a task temporarily during deletion or other operations, very well used!

dominic2412 commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,110 @@
+        }
+    }
+
+    public void addTask(String input) {
```
The variable input is well-named as it clearly indicates that it holds user input. It is appropriately concise and descriptive.

dominic2412 commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,110 @@
+    public void addTask(String input) {
+        Task task = new TodoTask("");
+        if(input.startsWith("todo ")) {
+            String descrip = input.substring(5).trim();
```
The abbreviation descrip is a bit unclear at first glance. It would be better to use the full word description for clarity.

dominic2412 reviewed (commented):
In summary, your code largely follows good naming conventions, which significantly contributes to its readability and maintainability. The main areas for improvement are in using more descriptive class names and avoiding abbreviations like descrip. These changes will make your code even more intuitive and easier for others to understand.

Keep up the good work! :))

adipanda2002 commented on src/main/java/Maga.java:
```diff
@@ -0,0 +1,40 @@
+import task.TaskList;
+import task.TaskManager;
+
+public class Maga {
```
I like your use of OOP concepts to make your main class clean and easy to read

adipanda2002 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,80 @@
+                continue;
+            }
+
+            // mark tasks as done
```
comments to explain each command follow coding style well and helpful in understanding the code

adipanda2002 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,80 @@
+    }
+
+    public void handleInput(String input) {
+        while(!input.equalsIgnoreCase("bye")) {
```
You could consider the use of enums instead of representing Tasks as Strings.

adipanda2002 commented on src/main/java/Maga.java:
```diff
@@ -0,0 +1,40 @@
+    private TaskList initialiseBot(TaskManager taskManager) {
+        return taskManager.loadTasks();
+    }
+
```
I think you are missing Javadoc commands across your codebase? I understand it is time consuming, but perhaps writing javadocs for at least the key methods will enable future readers of your code to understand it more seamlessly.

adipanda2002 commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,59 @@
+            String to = parts[4];
+            return new DeadlineTask(isDone, description, from, to);
+        }
+
```
Did you mean to have 'from' and 'to' parameters for Deadline and 'dateTime' for Event? Logically speaking, they should be the other way around don't you think?

adipanda2002 commented on src/main/java/task/TaskManager.java:
```diff
@@ -0,0 +1,50 @@
+    public TaskList loadTasks() {
+        TaskList tasks = new TaskList(100);
+        File file = new File(FILE_PATH);
+
```
I like that you are sticking to the style when it comes to indenting comments relative to their position in the code

adipanda2002 reviewed (commented):
Overall, LGTM. Your OOP principles are generally well followed in your code. I feel the readability of your code could be enhanced by adding in Javadoc comments, and also perhaps using enums. Some other nits to fix here and there.

audipras committed:
Delete unused Duke.java file

audipras committed:
Merge branch 'master' into add-gradle-support

audipras committed:
Add gradle wrapper + Add wrapping duke package

audipras committed:
Fix build.gradle mainclass directory

audipras committed:
Refactor code structure

Add Command, Controller and InvalidCommandException classes
Add Junit Tests for delete, mark and unmark functions

audipras committed:
Add find function + checkstyle

audipras committed:
Add java doc for most classes and public methods

audipras committed:
Merge branch 'branch-A-JavaDoc'

audipras committed:
Fix off by 1 JUnit test bug

audipras committed:
Amend code snippets to follow Java Coding Standard

audipras committed:
Fix CheckStyle errors

audipras committed:
Add javafx ui

audipras committed:
Add close functionality + fix save functionality

Using the exit command now properly closes
javafx window after a pause.

Bot now properly saves tasks
to the task file after every change
to the task list.

audipras committed:
Package classes

Create packages 'ui', 'middleware', 'commands'
and 'exceptions' to further organise code.

audipras committed:
Refactor package imports

audipras committed:
Delete text-ui test

audipras committed:
Add new exception

audipras committed:
Merge pull request #2 from audipras/branch-A-CodeQuality

Add new exception

audipras committed:
Add assertions

audipras committed:
Merge branch 'master' into branch-A-Assertions

audipras committed:
Merge pull request #3 from audipras/branch-A-Assertions

Add assertions

audipras committed:
Add tagging functionality

Tagging tasks is unavailable.

User can't categorise tasks since description can't be changed.

Add new 'tag' string field to each task, and add new 'tag' command so
users can tag current tasks.

audipras committed:
Merge branch 'branch-C-Tagging'

audipras committed:
Add user guide

No help manual for the user.

Create user guide to orient the user on how to use the bot.

The user can immediately use the bot without facing errors with bot
usage.

audipras committed:
Fix Readme Grammar

audipras committed:
Update javadocs to include tag field

audipras committed:
Fix checkstyle errors

audipras committed:
Fix delete test case

audipras committed:
Add local txt file to .gitignore
